src/Database/IdentifierInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.4.1
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database;
+
+/**
+ * Represents a single identifier name in the database
+ */
+interface IdentifierInterface extends ExpressionInterface
```

Comment:
I don't think there is any context in the query builder or SQL drivers where either needs to know an expression is an identifier. With the way it's designed those shouldn't care. An expression is an expression and they generate SQL. The only value in having an IdentifierInterface is so that you can change the value of the identifier later, which has no use case if you ask me.